std/path.d

```diff
@@ -3902,6 +3902,58 @@ if (isConvertibleToString!Range)
     assert(testAliasedString!isValidPath("/foo/bar"));
 }
 
+/** Checks whether `path1` is a subpath of `path2`
+  *
+  * Params:
+  *     path1 = string or Range of characters
+  *     path2 = string or Range of characters
+  *
+  * Returns:
+  *     true if `path` is a subpath of `path2`
+  *     false otherwise
+*/
+bool isSubPath(Range)(Range subPath, Range path)
+if ((isRandomAccessRange!Range && hasLength!Range && hasSlicing!Range &&
+    isSomeChar!(ElementEncodingType!Range) || isNarrowString!Range) &&
+    !isConvertibleToString!Range)
+{
+    import std.array : array;
+    
+    if (!isValidPath(subPath) || !isValidPath(path)) return false;
+
+    auto subPathAbsolute = asAbsolutePath(subPath).array;
+    auto pathAbsolute = asAbsolutePath(path).array;
+    subPathAbsolute = buildNormalizedPath(subPathAbsolute);
+    pathAbsolute = buildNormalizedPath(pathAbsolute);
```

Comment:
Are all these transformations necessary?

`asAbsolutePath`, in particular, makes the result depend on the current directory (which also precludes this function from being `pure`). I think this (whether to expand or not relative paths based on the current directory) could go either way, but, we ought to rationalize the decision.

Reply:
My recommendation is to require that path expansion is explicitly done on the caller side.

Instead, do this:
- If both paths are absolute, just compare the prefix (plus dir separator).
- If both paths are relative, just compare the prefix (plus dir separator).
- If only one of the paths is relative, assert.